src/main/scala/za/co/absa/commons/spark/SchemaUtils.scala

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2020 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.commons.spark
+
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types.{ArrayType, StructField, StructType}
+import org.apache.spark.sql.{Column, DataFrame}
+
+object SchemaUtils {
+  /**
+   * Compares 2 array fields of a dataframe schema.
+   *
+   * @param array1 The first array to compare
+   * @param array2 The second array to compare
+   * @return true if provided arrays are the same ignoring nullability
+   */
+  @scala.annotation.tailrec
+  private def isSameArray(array1: ArrayType, array2: ArrayType): Boolean = {
```

Comment:
I would rather use the word `equal` or `equivalent` instead of `same` for this purpose. The word `same` implies physical identity - the same object or the same reference.
```scala
equalArrayTypes(...)
```